main.go

```diff
@@ -95,7 +96,7 @@ func main() {
 		for f := range ch {
 			wg.Add(1)
 			go func(f *file) {
-				err := addLicense(f.path, f.mode, t, data)
+				err := addLicense(f.path, f.mode, t, data, *verbose)
```

Comment:
I wonder if it's better to do the verbose logging here for two reasons:
- cluttering arguments: the addLicense func never logs an error so why should it log anything else
- there's already if err != nil { ... } right below this line to print an error. It feels like log.Printf("%v modified", path) belongs here. So, maybe something like:

```go
defer wg.Done() // move it at the top
err := addLicense(f.path, f.mode, t, data)
if err != nil {
  log.Printf("%s: %v", f.path, err)
  return
}
if verbose {
  log.Printf("%s modified", path)
}
```

This also avoid making invalid claim "path/to/file modified" it there were an I/O error.

Reply:
Because if this function does not return an error, it doesn't mean the file has been modified. Specifically `if err != nil || hasLicense(b)` will return nil if we consider that the file already has a license.
We could change the definition of addLicense to return (modified bool, err error).

Reply:
ah sorry you're right!
yeah, I like the idea of `addLicense(...) (modified bool, err error)`.
let's do that?

Reply:
Done. PTAL. Thanks.